src/backend/api/misc.ts

```diff
@@ -83,12 +83,18 @@ export const clipboardReadText = async () =>
 export const clipboardWriteText = async (text: string) =>
   ipcRenderer.send('clipboardWriteText', text)
 
-export const handleShowErrorDialog = (
-  onError: (e: Electron.IpcRendererEvent, title: string, error: string) => void
+export const handleShowDialog = (
+  onMessage: (
+    e: Electron.IpcRendererEvent,
+    title: string,
+    message: string,
+    isError: boolean,
+    buttons: Array<string>
```

Comment:
Maybe we add here a map instead just the button names. Maybe we need to map a action later to each button. So each entry is a pair of the button name and the callback/action to trigger on click.

Reply:
We need to be able to specify the order the buttons will appear though. If we have a map like `{button_a: ()=>{}, etc.}` and we iterate through the keys, we won't be able to specify the order.